Sources/inc/STMTestBeamHeaders.hh

```diff
@@ -0,0 +1,90 @@
+#ifndef DataProducts_STMTestBeamHeaders_hh
+#define DataProducts_STMTestBeamHeaders_hh
+//
+// Header definitions for STM test beam data
+//
+
+// C++ includes
+#include <iostream>
+#include <vector>
+#include <array>
+#include <ctime>
+#include <iomanip>
+
+namespace mu2e {
+
+  namespace STMTestBeam {
+    struct TriggerHeader {
+      uint16_t deadbeef[2]; // fixed header word (= 0xDEADBEEF)
+      uint16_t datasize[2]; // data size
+      uint16_t n_slices[2]; // number of slices
+      uint16_t trigger_number[2]; // trigger number
+      uint16_t trigger_mode[1]; // trigger mode
+      uint16_t trigger_time[4]; // trigger time
+      uint16_t trigger_offset[2]; // time of first ADC value relative to trigger time
+      uint16_t n_dropped_packets[1]; // number of dropped packets
+
+      uint32_t getFixedHeader() const {
+        return ( uint32_t(deadbeef[1]) << 16 | uint32_t(deadbeef[0]) );
+      }
+
+      uint32_t getDataSize() const {
+        return ( uint32_t(datasize[1]) << 16 | uint32_t(datasize[0]) );
+      }
+
+      uint32_t getNSlices() const {
+        return ( uint32_t(n_slices[1]) << 16 | uint32_t(n_slices[0]) );
+      }
+
+      uint32_t getTriggerNumber() const {
+        return ( uint32_t(trigger_number[1]) << 16 | uint32_t(trigger_number[0]) );
+      }
+
+      uint16_t getTriggerMode() const {
+        return uint16_t(trigger_mode[0]);
+      }
+
+      uint64_t getTriggerTime() const {
+        return ( uint64_t(trigger_time[3]) << 48 | uint64_t(trigger_time[2]) << 32 | uint64_t(trigger_time[1]) << 16 | uint64_t(trigger_time[0]) );
+      }
+
+      uint32_t getTriggerOffset() const {
+        return ( uint32_t(trigger_offset[1]) << 16 | uint32_t(trigger_offset[0]) );
+      }
+
+      uint16_t getNDroppedPackets() const {
+        return uint16_t(n_dropped_packets[0]);
+      }
+
+      bool checkFixedHeader() const {
+        return (this->getFixedHeader() == (uint32_t) 0xDEADBEEF);
```

Comment:
I prefer that you use c++ style casts, not c style.  So uint32_t(0xDEADBEEF).  There are a few other places where you do this.  Please fix them all.

Reply:
Done in [128b677](https://github.com/Mu2e/Offline/pull/856/commits/128b677e431aaf6953d160d90ef0a47f6c9383ae). I couldn't find a way to do a c++-style cast for the (char*)s that I use in FromSTMTestBeamData_source.cc though

Reply:
I tried static_cast and dynamic_cast. Let me see if reinterpret_cast works

Reply:
reinterpret_cast worked. Done in 8e33f68eb